### PR TITLE
[BUG] Ensure `MultiRocketMultivariate` uses `random_state`

### LINKED ISSUE #5696: [BUG] Irreproducible results with `MultiRocketMultivariate`
`random_state` does guarantee the same results for each run.

```python
rng = np.random.default_rng()

X = pd.DataFrame([
    pd.Series([
        pd.Series(rng.integers(0, 10, 100)).astype(float),
        pd.Series(rng.integers(0, 10, 100)).astype(float),
    ]),
    pd.Series([
        pd.Series(rng.integers(0, 10, 100)).astype(float),
        pd.Series(rng.integers(0, 10, 100)).astype(float),
    ]),
])

MultiRocketMultivariate(random_state=42, num_kernels=84).fit_transform(X) - MultiRocketMultivariate(random_state=42, num_kernels=84).fit_transform(X)
```

The output should always be a `DataFrame` of zeros, but this is not the case.

<details>

System:
    python: 3.9.6 (default, Aug 11 2023, 19:44:49)  [Clang 15.0.0 (clang-1500.0.40.1)]
executable: /Users/temp/sktime/venv/bin/python
   machine: macOS-14.1.2-arm64-arm-64bit

Python dependencies:
          pip: 23.3.1
       sktime: 0.25.0
      sklearn: 1.3.2
       skbase: 0.4.6
        numpy: 1.26.1
        scipy: 1.11.3
       pandas: 2.1.4
   matplotlib: 3.8.1
       joblib: 1.3.2
        numba: 0.58.1
  statsmodels: 0.14.0
     pmdarima: None
statsforecast: None
      tsfresh: 0.20.1
      tslearn: 0.6.3
        torch: 2.1.0
   tensorflow: None
tensorflow_probability: None

</details>

### PR DESCRIPTION
**Pull request by chrico-bu-uab**
#### Reference Issues/PRs

Fixes #5696.

#### What does this implement/fix? Explain your changes.

Sets the random state.

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### What should a reviewer concentrate their feedback on?

N/A

#### Did you add any tests for the change?

N/A

#### Any other comments?

Have not yet added myself to the list/file (which one?), will do later.

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks!

Though, afaik, `seed` is deprecated and generators should be used instead:
https://numpy.org/doc/stable/reference/random/generated/numpy.random.seed.html

**Review by fkiraly — APPROVED** [review]
Let's merge this now, we can address random seeding more systematically.
